components/org.wso2.carbon.identity.idle.account.identification/src/main/java/org/wso2/carbon/identity/idle/account/identification/dao/impl/IdleAccIdentificationDAOImpl.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.idle.account.identification.dao.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.idle.account.identification.constants.IdleAccIdentificationConstants;
+import org.wso2.carbon.identity.idle.account.identification.dao.IdleAccIdentificationDAO;
+import org.wso2.carbon.identity.idle.account.identification.exception.IdleAccIdentificationException;
+import org.wso2.carbon.identity.idle.account.identification.exception.IdleAccIdentificationServerException;
+import org.wso2.carbon.identity.idle.account.identification.models.InactiveUserModel;
+import org.wso2.carbon.identity.idle.account.identification.util.Utils;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation class for IdleAccIdentificationDAO.
+ */
+public class IdleAccIdentificationDAOImpl implements IdleAccIdentificationDAO {
+
+    @Override
+    public List<InactiveUserModel> getInactiveUsers(String inactiveAfter, String excludeBefore, String tenantDomain)
+            throws IdleAccIdentificationException {
+
+        if (StringUtils.isEmpty(excludeBefore)) {
+            return getInactiveUsersFromSpecificDate(inactiveAfter, tenantDomain);
+        } else {
+            return getLimitedInactiveUsersFromSpecificDate(inactiveAfter, excludeBefore, tenantDomain);
+        }
```

Comment:
```suggestion

        return getLimitedInactiveUsersFromSpecificDate(inactiveAfter, excludeBefore, tenantDomain);

```

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/698/commits/891e0922ab1f1a685f3049500ff4ab014bdec860